lib/command.js

```diff
@@ -97,6 +130,152 @@ export async function processCommand({nick, message, participants, say}) {
     process.kill(process.pid, 'SIGINT');
     return;
   }
+  if(command == 'create' && args[0] === 'poll') {
+    args.shift();
+    const cmd = args.join(' ');
+    const createArgs = cmd.split(/\s*\|\s*/).filter(Boolean);
+    if(createArgs.length < 3) {
+      say('Try: create poll QUESTION | OPTION 1 | OPTION 2 ...');
+      return;
+    }
+    const question = createArgs.shift();
+    const answers = createArgs;
+    const senderId = polls.voterIdsByNick[nick] ||
+      (polls.voterIdsByNick[nick] = nonCryptographicHash8(nick));
+    const newPollMsg = {
+      type: 'new-poll',
+      pollId: nonCryptographicHash8(cmd),
+      senderId,
+      senderName: nick,
+      question,
+      answers
+    };
+    await msg(newPollMsg);
+    return;
+  }
+  if(command === 'vote') {
+    // "vote [option[s]] <number>[,...] [on [poll] [#]<number>]"
+    if(args[0] === 'option' || args[0] === 'options') {
+      args.shift();
+    }
+    if(args.length === 0) {
+      say('Vote what option on what poll?');
+      return;
+    }
+    let options = args.shift().split(/,\s*/);
+    if(options.some(isNaN)) {
+      say(`Expected numeric options (e.g. 1,2) instead of "${options}"`);
+      return;
+    }
+    let pollNum;
+    if(args.length > 0) {
+      if(args[0] === 'on') {
+        args.shift();
+      } else {
+        say('Try: vote option X on poll Y');
+        return;
+      }
+      if(args[0] === 'poll') {
+        args.shift();
+      }
+      if(args[0]) {
+        pollNum = args.shift().replace(/^#/, '');
+      }
+    }
+    const poll = pollNum
+      ? polls.byId[polls.idsByNum[pollNum]]
+      : polls.byId[polls.idsByNum[polls.count]];
+    if(!poll) {
+      say(`Unknown poll "${pollNum}"`);
+      return;
+    }
+    const answersMap = {};
+    for(const option of options) {
+      const answerI = option-1;
+      answersMap[answerI] = true;
+      const answer = poll.msg.answers[answerI];
+      if(answer == null) {
+        say(`Unknown option "${option}"`);
```

Comment:
People might get confused here since IRC is async, so people's votes might not get counted and they might not realize?

Reply:
Do you mean in this particular error (unknown option chosen)? Or in all the error/failure cases for this command?
The responses could say something additional like "vote not processed, try again" to be more explicit about the vote not being counted? Or the response could be targeted to the user (maybe using `<nick>, message` since `nick: message` might get interpreted as scribing?) so they would be more likely to notice that it is relevant to them. There could also be unhandled vote attempts that are not matched at all by this handler and so would not generate a response. But a successful vote will result in a IRC message saying that the answer/vote took place, as generated by the bot when it receives back its own `answer-poll` xmpp message, and that (along with the vote appearing in the Jitsi UI) is the positive confirmation.

Reply:
Changed to list available options if an unknown option is picked: 2bb6c537a682506bb1573e458addd516208c706f
```
22:14 <@cel> cgbot-cel: create poll Test? | Yes | No
22:14 <cgbot-cel> cel created a poll #1 with 2 options.
22:14 <cgbot-cel> POLL: Test?
22:14 <cgbot-cel> ... Option 1: Yes
22:14 <cgbot-cel> ... Option 2: No
22:14 <@cel> cgbot-cel: vote 3
22:14 <cgbot-cel> Unknown option "3" on poll #1. Available options: 1,2
```